10/RequestReply.md

```diff
@@ -127,6 +127,62 @@ Operations case:
+
+When a Client activates a Device Lock, the Server MUST reject any of the following request from other Client:
+* Command call except for State Command, Status Command and all commands listed in the Allowed Commands list (described further down).
+* Write Attribute Request (including the atomic Write Read request)
+* Write Pipe Request (including the atomic Write Read request)
+* Setting Attribute Configuration
+* Setting Pipe Configuration
+* Setting Polling Configuration
+* Setting Logging Configuration
+In general it is RECOMMANDED to block any request resulting a mutation of a Device.
+
+An Allowed Command is not affected by any the Device Lock, even this implies a modification of the Device state. The DeviceServer define a list of Allowed Command per Device Class which can only be changed at the Initialisation phase (TODO find the exact term) of the Device Server.
+TODO: Explain how to set/get this list?
+
+Other Clients MAY use DevLockStatus command of a DeviceServer to check if a Device Lock is currently activated.
+
+The request of a Device Lock is done throught the DeviceServer command LockDevice. The protocol of activation SHALL be the responsibility of the Device Server, following this sequence: (?)
+* Check no other Client owns an active and valid DeviceLock on the requested Device
+* Activate the Device Lock 
+* Increase the counter of Device Lock (See Below).
+* Case of Forwarded Attribute(?): a Device Lock request SHALL be sent and validated by all Devices connected to the Forwarded Attributes of the targetted Device. 
+In case of failure of the activation of a Device Lock the Device Server SHALL throw a related DevFailed exception with \<reason\> field set to "API_DeviceLocked".
+
+A Client owning a Device Lock MAY:
+* revoke the Device Lock by the DeviceServer command UnLockDevice,
+* renew the Device Lock activation by the DeviceServer command ReLockDevices
+A Locking Tango Exception SHALL be thrown if the Device Lock is not active anymore.
+
+
+A Device Lock is defined by:
+* a Client Identification (TODO specify in this RFC) of the owner
+* the Client Identification of the previous owner (Mandatory?)
+* a timestamp corresponding of the locking activation
+* a counter (?) representing the number of valid activation from the same Client 
+* the activation time in second defining the locking period from the last activation timestamp  # time_t in the implementation?)
+
+A Device Lock SHALL be deactivated when:
+* The Client which locked the Devcie call DeviceServer command UnLockDevice for the locked Device,
+* or Time of activation expired,
+* or the Connection to the Client is lost,
+* The Device Server is reinitialised.
+
+A Client Identication SHOULD cary these information:
+* Hostname where the Client process runs
+* PID of the Client process
+* IP of the Client connection
+It is recommended to include any other informations which may help the administrators of the system.  
```

Comment:
I am not sure if we should specify this here, as the current implementation differ in this informations for c++ and cpp clients.